stdlib/email/src/main/java/org/ballerinalang/stdlib/email/client/PopClient.java

```diff
@@ -0,0 +1,123 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.NoSuchProviderException;
+import javax.mail.Session;
+import javax.mail.Store;
+
+/**
+ * Contains functionality of POP Client.
+ */
+public class PopClient {
+
+    private static final Logger log = LoggerFactory.getLogger(PopClient.class);
+
+    private PopClient() {
+        // A Singleton class.
+    }
+
+    /**
+     * Initializes the ObjectValue object with the POP Properties.
+     * @param clientEndpoint Represents the POP Client class
+     * @param config Properties required to configure the POP Session
+     * @param isPop True if the protocol is POP3 and false otherwise (if protocol is IMAP)
+     * @throws BallerinaPopException If an error occurs in the POP client
+     */
+    public static void initClientEndpoint(ObjectValue clientEndpoint, MapValue<Object, Object> config, boolean isPop)
+            throws BallerinaPopException {
+        log.debug("[PopClient][InitClient] Calling getProperties");
+        Properties properties = PopUtil.getProperties(config, isPop);
+        Session session = Session.getInstance(properties, null);
+        try {
+            Store store;
+
+            if (isPop) {
+                store = session.getStore(PopConstants.POP_PROTOCOL);
+                clientEndpoint.addNativeData(PopConstants.PROPS_HOST,
+                        properties.getProperty(PopConstants.PROPS_POP_HOST));
+            } else {
+                store = session.getStore(ImapConstants.IMAP_PROTOCOL);
+                clientEndpoint.addNativeData(PopConstants.PROPS_HOST,
+                        properties.getProperty(ImapConstants.PROPS_IMAP_HOST));
+            }
+            clientEndpoint.addNativeData(PopConstants.PROPS_STORE, store);
+            clientEndpoint.addNativeData(PopConstants.PROPS_USERNAME,
+                    properties.getProperty(PopConstants.PROPS_USERNAME));
+            clientEndpoint.addNativeData(PopConstants.PROPS_PASSWORD,
+                    properties.getProperty(PopConstants.PROPS_PASSWORD));
+        } catch (NoSuchProviderException e) {
+            log.error("Failed to read message : ", e);
+            throw new BallerinaPopException("Error occurred while accessing POP server", e);
+        }
+    }
+
+    /**
+     * Read emails from the server.
+     * @param clientConnector Represents the POP Client class
+     * @param filter Criteria, which is used to read emails
+     * @param isPop True if the protocol is POP3 and false otherwise (if protocol is IMAP)
+     * @return MapValue Returns the type supported by Ballerina
+     * @throws BallerinaPopException If an error occurs in the POP client
+     */
+    public static MapValue readMessage(ObjectValue clientConnector, MapValue<Object, Object> filter, boolean isPop)
```

Comment:
Seems like the `isPop` method parameter is not used. Please check all the usages of this method.
```suggestion
    public static MapValue readMessage(ObjectValue clientConnector, MapValue<Object, Object> filter)
```

Reply:
Removed from this location and the references.